src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -376,13 +359,12 @@ public SAMRecordIterator assertSorted(final SortOrder sortOrder) {
             seekableStream.seek(startOfLastLinearBin >>> 16);
             final Container container = ContainerIO.readContainerHeader(newIterator.getCramHeader().getVersion().major, seekableStream);
             seekableStream.seek(seekableStream.position() + container.containerByteSize);
-            iterator = newIterator;
-            iterator.jumpWithinContainerToPos(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, SAMRecord.NO_ALIGNMENT_START);
+            newIterator.jumpWithinContainerToPos(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, SAMRecord.NO_ALIGNMENT_START);
+            return newIterator;
```

Comment:
Wait, wont this leave this in a bad state, namely `iterator` is no longer set correctly?